packages/i18n/src/create-i18n.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import Tannin from 'tannin';
+
+/**
+ * @typedef {Record<string,any>} LocaleData
+ */
+
+/**
+ * Default locale data to use for Tannin domain when not otherwise provided.
+ * Assumes an English plural forms expression.
+ *
+ * @type {LocaleData}
+ */
+const DEFAULT_LOCALE_DATA = {
+	'': {
+		plural_forms: ( n ) => ( n === 1 ? 0 : 1 ),
+	},
+};
+
+/**
+ * An i18n instance
+ *
+ * @typedef {Object} I18n
+ * @property {Function} setLocaleData Merges locale data into the Tannin instance by domain. Accepts data in a
+ *                                    Jed-formatted JSON object shape.
+ * @property {Function} __            Retrieve the translation of text.
+ * @property {Function} _x            Retrieve translated string with gettext context.
+ * @property {Function} _n            Translates and retrieves the singular or plural form based on the supplied
+ *                                    number.
+ * @property {Function} _nx           Translates and retrieves the singular or plural form based on the supplied
+ *                                    number, with gettext context.
+ * @property {Function} isRTL         Check if current locale is RTL.
+ */
+
+/**
+ * Create an i18n instance
+ *
+ * @param {LocaleData} [initialData]    Locale data configuration.
+ * @param {string}     [initialDomain]  Domain for which configuration applies.
+ * @return {I18n}                       I18n instance
+ */
+export const createI18n = ( initialData, initialDomain ) => {
+	/**
+	 * The underlying instance of Tannin to which exported functions interface.
+	 *
+	 * @type {Tannin}
+	 */
+	const tannin = new Tannin( {} );
+
+	/**
+	 * Merges locale data into the Tannin instance by domain. Accepts data in a
+	 * Jed-formatted JSON object shape.
+	 *
+	 * @see http://messageformat.github.io/Jed/
+	 *
+	 * @param {LocaleData} [data]   Locale data configuration.
```

Comment:
It's not optional.

```suggestion
	 * @param {LocaleData} data     Locale data configuration.
```

Related:

- https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#nullable-undefined-and-void-types
- https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

Reply:
This becomes a type error:

https://github.com/WordPress/gutenberg/blob/c0e49d9fe4a44a63f64acc0ab9bf819d2fa9a9ff/packages/i18n/src/create-i18n.js#L99

Do you have a suggestion on how to proceed?

Reply:
I guess it may be the equivalent to pass an empty object there, but it demonstrates a likely-legitimate case for passing an `undefined` data set.

If it's easiest, we can just make it optional again.